composition-js/src/validate.ts

```diff
@@ -216,17 +215,40 @@ function generateWitnessValue(type: InputType): any {
   }
 }
 
-export function validateGraphComposition(supergraph: QueryGraph, subgraphs: QueryGraph): {errors? : ValidationError[]} {
-  const errors = new ValidationTraversal(supergraph, subgraphs).validate();
+/**
+ * Validates that all the queries expressable on the API schema resulting of the composition of the provided subgraphs can be executed
+ * on those subgraphs.
+ *
+ * @param supergraphSchema the schema of the supergraph that composing `subgraphs` generated. Note this *must* be the full supergraph, not
+ *   just it's API schema (because it may be used to find the definition of elements that are marked `@inaccessible`). Note that this _not_
+ *   the same schema that the one reference inside `supergraphAPI` in particular.
+ * @param supergraphAPI the `QueryGraph` corresponding to the `supergraphSchema` API schema.
+ * @param subgraphs the (federated) `QueryGraph` corresponding the subgraphs having been composed to obtain `supergraphSchema`.
+ */
+export function validateGraphComposition(
+  supergraphSchema: Schema,
+  supergraphAPI: QueryGraph,
+  subgraphs: QueryGraph
+): {
+  errors? : ValidationError[]
+} {
+  const errors = new ValidationTraversal(supergraphSchema, supergraphAPI, subgraphs).validate();
   return errors.length > 0 ? {errors} : {};
 }
 
-export function computeSubgraphPaths(supergraphPath: RootPath<Transition>, subgraphs: QueryGraph): {traversal?: ValidationState, isComplete?: boolean, error?: ValidationError} {
+export function computeSubgraphPaths(
```

Comment:
Note to reviewer: if you look up, you may notice that this method is not used internally. But it's mean to be a handy tool to help validation at times (it's essentially about walking validation step-by-step).